source/client-side-encryption/client-side-encryption.rst

```diff
@@ -477,7 +479,7 @@ accept arbitrary strings at runtime for forward-compatibility.
 .. code:: typescript
 
    interface KMSProviders {
-      aws?: AWSKMSOptions;
+      aws?: AWSKMSOptions | { /* Empty. (See "Automatic AWS Credentials") */ };
```

Comment:
JAVA-4499 also adds [KMS suppliers](https://github.com/mongodb/mongo-java-driver/commit/e9a555b6fd89e2d457e575e4b296f33558e83cf6#diff-66335290fb5cf287cabdef482d747b9bb16c8b948f45dc2e68001bbcb856210aR134) for all KMS providers. Not limiting to AWS may provide workarounds for obtaining credentials in more ways for other KMS providers (e.g. GCP has [several sources for credentials](https://google.aip.dev/auth/4110)).

I am assuming that is included in the scope of DRIVERS-2280. I left a comment to confirm. Assuming that is OK, I recommend making these changes non-AWS specific, and adding the `KMSSuppliers` map option.

Reply:
I have seen this comment, but still holding off for further clarification on the scope.

Reply:
Comments in DRIVERS-2280 clarified this is not in scope. Resolving.